modules/API/pages/built-in-endpoints.adoc

```diff
@@ -3384,6 +3384,144 @@ Response::
 | String ID of the query.
 |===
 
+
+=== Get information about a template query
+
+`GET /gsql/library?functionName=pkg1.pkg2.<template name>`
+
+This endpoint returns the query definition, query name, and query parameters for a template query.
+
+==== Sample request
+[tabs]
+====
+Request::
++
+--
+[source.wrap,bash]
+----
+curl -X GET --user tigergraph:tigergraph --url 'http://localhost:14240/gsqlserver/library?functionName=gds.community.louvain'
+----
+--
+Response::
++
+--
+[source,json]
+----
+{
+  "error": false,
+  "message": "",
+  "results": [
+    {
+      "query": <GSQL definition of the Louvain algorithm, omitted for length>,
+      "name": "louvain",
+      "params": {
+        "file_path": {
+          "default": "",
+          "type": "STRING",
+          "min_count": 0
+        },
+        <additional parameters for the Louvain algorithm, omitted for length>
+        "e_type": {
+          "max_count": 2147483647,
+          "type": "STRING",
+          "min_count": 0
+        }
+      }
+    }
+  ]
+}
+----
+--
+====
+==== Parameters
+
+|===
+| Name | Required | Description
+
+| `functionName`
+| Yes
+| The template name that will be returned in the response
+
+| `isRegularExpression`
+| No
+| Enables regular expression searching for the function name. For example, `gds.*` returns all queries that begin with `gds`.
+|===
+
+
+=== Run a template query
+
+`POST /gsql/library?graph=<graph name>&functionName=pkg1.pkg2.templateName`
+
+This endpoint runs a template query on a given graph.
+Query parameters are passed in JSON format in the body of the POST request.
+
+==== Sample request
+[tabs]
+====
+Request::
++
+--
+[source.wrap,bash]
+----
+curl -X POST --user tigergraph:tigergraph \
+  --url 'http://127.0.0.1:8123/gsql/library?functionName=gds.community.louvain&graph=poc_graph' \
+  --data '{
+            "v_type": [
+              "members"
+            ],
+            "e_type": [
+              "member_work_company"
+            ],
+            "wt_attr": "positionId",
+            "max_iter": 10,
+            "result_attr": "",
+            "file_path": "",
+            "print_info": true
+          }'
+----
+--
+Response::
++
+--
+[source,json]
+----
+{
+  "error": false,
+  "message": "",
+  "results": {    <--- query's response
+    "error": false,
+    "message": "",
+    "version": {
+      "schema": 1,
+      "edition": "enterprise",
+      "api": "v2"
+    },
+    "results": [
+      {"@@comp_group_by_size_map": {}},
+      {"sizes": {}},
+      {"Start": []}
+    ]
+  },
+  "generatedQueryName": "gds_community_louvain_1502296930"
+}
+----
+--
+====
+==== Parameters
+
+|===
+| Name | Required | Description
+
+| `functionName`
+| Yes
+| The template name that will be returned in the response
```

Comment:
The description of this field you wrote is not correct enough.
should be the one I wrote.

Reply:
Sorry, I mistyped which section I put your description in.